src/nomad_measurements/xrd/nx.py

```diff
@@ -37,3 +47,86 @@
     '/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/ratio_k_alphatwo_k_alphaone',
     '/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/kbeta',
 ]
+
+
+CONCEPT_MAP = {
+    '/ENTRY[entry]/@default': 'experiment_result',
+    '/ENTRY[entry]/definition': 'NXxrd_pan',
+    '/ENTRY[entry]/method': 'archive.data.method',
+    '/ENTRY[entry]/measurement_type': 'archive.data.diffraction_method_name',
+    '/ENTRY[entry]/experiment_result/@signal': 'intensity',
+    '/ENTRY[entry]/INSTRUMENT[instrument]/DETECTOR[detector]/scan_axis': 'archive.data.results[0].scan_axis',
+    '/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/xray_tube_material': 'archive.data.xrd_settings.source.xray_tube_material',
+    '/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/xray_tube_current': 'archive.data.xrd_settings.source.xray_tube_current',
+    '/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/xray_tube_voltage': 'archive.data.xrd_settings.source.xray_tube_voltage',
+    '/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/k_alpha_one': 'archive.data.xrd_settings.source.kalpha_one',
+    '/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/k_alpha_two': 'archive.data.xrd_settings.source.kalpha_two',
+    '/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/ratio_k_alphatwo_k_alphaone': 'archive.data.xrd_settings.source.ratio_kalphatwo_kalphaone',
+    '/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/kbeta': 'archive.data.xrd_settings.source.kbeta',
+}
+
+
+def walk_through_object(parent_obj, attr_chain):
+    """
+    Walk though the object until reach the leaf.
+
+    Args:
+        parent_obj: This is a python obj.
+            e.g.Arvhive
+        attr_chain: Dot separated obj chain.
+            e.g. 'archive.data.xrd_settings.source.xray_tube_material'
+        default: A value to be returned by default, if not data is found.
+    """
+    if parent_obj is None:
+        return parent_obj
+
+    if isinstance(attr_chain, str) and attr_chain.startswith('archive.'):
+        parts = attr_chain.split('.')
+        child_obj = None
+        for part in parts[1:]:
+            child_nm = part
+            if '[' in child_nm:
+                child_nm, index = child_nm.split('[')
+                index = int(index[:-1])
+                # section always exists
+                child_obj = getattr(parent_obj, child_nm)[index]
+            else:
+                child_obj = getattr(parent_obj, child_nm, None)
+            if child_obj is None:
+                return None
+            parent_obj = child_obj
+
+        return child_obj
+
+
+def populate_nx_dataset_and_attribute(
```

Comment:
This can be made part of the HDF5Handler, no?

Reply:
Done!